test/cartesian/CartesianGrid.spec.tsx

```diff
@@ -30,6 +30,179 @@ describe('<CartesianGrid />', () => {
     brushBottom: 7,
   };
 
+  const chartMargin: Margin = {
+    bottom: 11,
+    left: 12,
+    right: 13,
+    top: 14,
+  };
+
+  const expectedOffset: ChartOffset = {
+    bottom: 11,
+    brushBottom: 11,
+    height: 175,
+    left: 12,
+    right: 13,
+    top: 14,
+    width: 275,
+  };
+
+  describe('layout and size when set explicitly', () => {
+    it('should put x, y, width and height as coordinates to the background', () => {
+      const { container } = render(
+        <Surface width={500} height={500}>
+          <CartesianGrid
+            x={1}
+            y={2}
+            width={100}
+            height={200}
+            fill="green"
+            verticalPoints={verticalPoints}
+            horizontalPoints={horizontalPoints}
+          />
+        </Surface>,
+      );
+      const background = container.querySelector('rect.recharts-cartesian-grid-bg');
+      expect.soft(background).toHaveAttribute('x', '1');
+      expect.soft(background).toHaveAttribute('y', '2');
+      expect.soft(background).toHaveAttribute('width', '100');
+      expect.soft(background).toHaveAttribute('height', '200');
+    });
+
+    it('should put x, y, width and height as coordinates to all lines', () => {
+      const { container } = render(
+        <Surface width={500} height={500}>
+          <CartesianGrid
+            x={1}
+            y={2}
+            width={100}
+            height={200}
+            fill="green"
+            verticalPoints={verticalPoints}
+            horizontalPoints={horizontalPoints}
+          />
+        </Surface>,
+      );
+      const horizontalLines = container.querySelectorAll('.recharts-cartesian-grid-horizontal line');
+      expect(horizontalLines).toHaveLength(5);
+      horizontalLines.forEach(line => {
+        expect.soft(line).toHaveAttribute('x', '1');
+        expect.soft(line).toHaveAttribute('y', '2');
+        expect.soft(line).toHaveAttribute('x1', '1');
+        expect.soft(line).toHaveAttribute('x2', '101');
+      });
+      const verticalLines = container.querySelectorAll('.recharts-cartesian-grid-vertical line');
+      expect(verticalLines).toHaveLength(4);
+      verticalLines.forEach(line => {
+        expect.soft(line).toHaveAttribute('x', '1');
+        expect.soft(line).toHaveAttribute('y', '2');
+        expect.soft(line).toHaveAttribute('y1', '2');
+        expect.soft(line).toHaveAttribute('y2', '202');
+      });
+    });
+
+    it('should put x, y, width and height as coordinates to all stripes', () => {
+      const { container } = render(
+        <Surface width={500} height={500}>
+          <CartesianGrid
+            x={1}
+            y={2}
+            width={100}
+            height={200}
+            fill="green"
+            verticalPoints={verticalPoints}
+            horizontalPoints={horizontalPoints}
+            horizontalFill={['red', 'black']}
+            verticalFill={['green', 'blue']}
```

Comment:
The fill seems unrelated to the test case.

Reply:
This is required because without `horizontalFill` and `verticalFill`, CategoricalChart will not render the stripes

Reply:
Thank you for the context, I was not aware. This is helpful knowledge. ❤️